aten/src/ATen/native/mps/operations/ReduceOps.mm

```diff
@@ -1847,11 +1847,31 @@ Tensor median_mps(const Tensor& input_t) {
             MPSGraphTensor * reshapedTensor = [mpsGraph reshapeTensor:inputTensor
                                                             withShape:@[@-1]
                                                                   name:nil];
+            MPSDataType dataType = [inputTensor dataType];
+            // #issue 104398441 sortWithTensor only supports following types, cast if necessary
+            if (dataType != MPSDataTypeInt32 &&
```

Comment:
Maybe you can use a tuple and then compare with it instead of repeating the same conditional here and below ?